packages/sdk/src/types/config.ts

```diff
@@ -0,0 +1,71 @@
+export enum Environment {
+  DEVNET = 'devnet',
+  TESTNET = 'testnet',
+  MAINNET = 'mainnet',
+}
+
+export type Domain = {
+  id: number;
+  name: string;
+};
+
+export enum ResourceType {
+  ERC20 = 'erc20',
+  ERC721 = 'erc721',
+  PERMISSIONED_GENERIC = 'permissionedGeneric',
+  PERMISSIONLESS_GENERIC = 'permissionlessGeneric',
+}
+
+export enum Network {
+  EVM = 'evm',
+  SUBSTRATE = 'substrate',
+}
+
+export type Resource = {
+  resourceId: string;
+  type:
+    | ResourceType.ERC20
+    | ResourceType.ERC721
+    | ResourceType.PERMISSIONED_GENERIC
+    | ResourceType.PERMISSIONLESS_GENERIC;
```

Comment:
If ResourceType were declared as a type with string literals, this type definition could be simplified to

```ts
export type ResourceType = "erc20" | "erc721" | "permissionedGeneric" | "permissionlessGeneric"

export type Resource = {
  resourceId: string,
  type: ResourceType
}
```